content/en/docs/reference/node/kubelet-files.md

```diff
@@ -0,0 +1,95 @@
+---
+content_type: "reference"
+title: Files written and read by the kubelet
+weight: 40
+---
+
+{{< glossary_tooltip text="Kubelet" term_id="kubelet" >}} is mostly a stateless
+process running on a Kubernetes {{< glossary_tooltip text="node" term_id="node" >}}.
+This document outlines files that kubelet reads and writes.
+
+Kubelet uses the {{< glossary_tooltip text="control plane" term_id="control-plane" >}} as
+the source of truth on what needs to run on the Node, and the
+{{<glossary_tooltip text="container runtime" term_id="container-runtime">}} to retrieve
+the current state of containers. Kubelet also relies on reading
+cgroups and various system files to collect metrics. However, there are still a few other files
+that are used by the kubelet.
+
+## Files
+
+### Configuration
+
+#### Kubelet configuration files
+
+The path to the kubelet configuration file can be configured
+using the command line argument `--config`. kubelet also supports
+[drop-in configuration files](/docs/tasks/administer-cluster/kubelet-config-file/#kubelet-conf-d)
+to enhance configuration.
+
+#### Certificates
+
+Certificates are typically located at `/var/lib/kubelet/pki`,
+but can be configured using the `--cert-dir` kubelet command line argument.
+Names of certificate files are also configurable.
+
+#### Systemd unit settings
+
+When kubelet is running as a systemd unit, some kubelet configuration may be declared
+in systemd unit settings file. Typically it includes:
+
+- command line arguments to [run kubelet](/docs/reference/command-line-tools-reference/kubelet/)
+- environment variables, used by kubelet or [configuring golang runtime](https://pkg.go.dev/runtime#hdr-Environment_Variables)
+
+### State
```

Comment:
Does Kubelet use any other files other than checkpoints to keep the state?

Reply:
no as far as I know. But please correct me if I am wrong

Reply:
I'd propose to remove this section as all files are already mentioned in the `Checkpoint files` section.

Reply:
ah, sorry, it's an upper level section.